agent/src/agent/api/routes/alerts.py

```diff
@@ -0,0 +1,187 @@
+    if not form.validate():
+        return jsonify({
+            'status': 'Abnormal request',
+            'errors': form.errors,
+        }), 400
+    try:
+        alert_groups = AnodotApiClient(destination.repository.get()).get_alerts({
+            'startTime': form.start_time.data,
+        })
+    except destination.repository.DestinationNotExists as e:
+        return jsonify({
+            'status': 'Not connected',
+            'errors': {'destination': [str(e)]},
+        }), 400
+    except requests.exceptions.HTTPError as e:
+        return _error_response(e)
+
+    _filter_groups_by_name(alert_groups, form.alert_name.data)
+    _filter_group_alerts_by_host(alert_groups, form.host.data)
+
+    alert_groups = alert_groups['alertGroups']
+    if len(alert_groups) == 0 or len(alert_groups[0]['alerts']) == 0:
+        return jsonify({'status': 'No alert'})
+    return jsonify(_extract_alert_statuses(alert_groups))
+
+
+@alerts.route('/alerts', methods=['GET'])
+def get_alerts():
+    form = AlertsByStatusForm(request.args)
+    if not form.validate():
+        return jsonify({
+            'status': 'Abnormal request',
+            'errors': form.errors,
+        }), 400
+    try:
+        params = {
+            'status': form.status.data,
+            'order': form.order.data,
+            'sort': form.sort.data,
+        }
+        # apply time constraints only to CLOSE status
+        if form.status.data != OPEN:
+            params['startTime'] = form.start_time.data
+        alert_groups = AnodotApiClient(destination.repository.get()).get_alerts(params)
+    except destination.repository.DestinationNotExists as e:
+        return jsonify({
+            'status': 'Not connected',
+            'errors': {'destination': [str(e)]},
+        }), 400
+    except requests.exceptions.HTTPError as e:
+        return _error_response(e)
+
+    _move_metric_dimensions(alert_groups)
+    _move_metric_scores(alert_groups)
+    return jsonify(_transform(alert_groups))
+
+
+def _error_response(e: requests.exceptions.HTTPError):
+    if e.response.status_code == 401:
+        return jsonify({
+            'status': 'Not connected',
+            'errors': {'Unauthorized': [str(e)]},
+        }), 401
+    return jsonify({
+        'status': 'Not connected',
+        'errors': {'HTTPError': [str(e)]},
+    }), 500
+
+
+def _filter_groups_by_name(alert_groups: dict, alert_name: str):
+    alert_groups['alertGroups'] = [group for group in alert_groups['alertGroups'] if group['name'] == alert_name]
+    _set_total(alert_groups)
+
+
+def _filter_group_alerts_by_host(alert_groups: dict, host: str):
+    to_delete = []
+    for i, group in enumerate(alert_groups['alertGroups']):
+        _filter_alerts_by_metric_host(group['alerts'], host)
+        if len(group['alerts']) == 0:
+            to_delete.append(i)
+    for i in to_delete:
+        del alert_groups['alertGroups'][i]
+
+
+def _filter_alerts_by_metric_host(alerts_: list, host: str):
```

Comment:
why are you deleting objects from the existing object instead of just returning the list of needed objects?

Reply:
if I return the list of needed objects - I'll set this list instead of the previous one, like
`alert_groups['alertGroups'] = _filter_alerts_by_metric_host(group['alerts'], host)`
so the result will be the same

Reply:
yeah, but you won't need to write 
```python
for i in to_delete:
        del smth
```